torch/_dynamo/variables/builder.py

```diff
@@ -1472,3 +1476,48 @@ def wrap_to_fake_tensor_and_record(
         return fake_e
     else:
         return e
+
+
+class SourcelessBuilder:
+    """
+    Like builder, but stateless and does not require a source. Useful for simple type->VT objects, or objects
+    that are being created/evaporated during inlining (ex: consider a locally made list of tensors we then iterate over
+    .), such a list should not show up as an artifact from inputs, nor in reconstruction, nor in the graph. However,
+    there may be reasons to represent it as a ListVariable internally.
+
+    NOTE - Objects produced here are born UNGUARDED due to the nature of sources!
+
+    NOTE - This class is very new! It will have some rough edges, but it was created to stem the bleeding of giant
+    if/else type->VariableTracker trees that were cropping up all over dynamo.
+    """
+
+    def __call__(self, tx, value) -> VariableTracker:
+        if isinstance(value, VariableTracker):
+            # This is always valid to call, and useful for recursive calls.
+            return value
+        if ConstantVariable.is_literal(value):
+            return SourcelessBuilder.wrap_constant_literal(value)
+        elif is_builtin_callable(value):
+            return BuiltinVariable(value)
+        elif is_allowed(value):
+            return TorchVariable(value)
+        elif isinstance(value, types.FunctionType):
+            return UserFunctionVariable(value)
+        elif isinstance(value, enum.Enum):
+            return EnumVariable(value)
+        elif isinstance(value, (type, abc.ABCMeta)):
+            return UserDefinedClassVariable(value)
+        elif isinstance(value, dict):
+            return ConstDictVariable(
+                {k: self(tx, v) for k, v in value.items()},
```

Comment:
whats the logic difference between calling `wrap_bound_arg` and SourcelessBuilder()?

If we create an ephemeral list/dict during inlining with SB, does that imply that the contents of the list should also be sourceless?  I thought it might be valid to build a new list from existing/external elements that need VT's built with real sources

Reply:
> whats the logic difference between calling wrap_bound_arg and SourcelessBuilder()?

I fixed some of the options piping to be correct

> If we create an ephemeral list/dict during inlining with SB, does that imply that the contents of the list should also be sourceless

If its a raw object, sourceless is the best we can do. It also means we've not seen it before. If it's not a raw object, it should be a VT, and therefore should have been seen already.

Reply:
I'll add a test.

Reply:
Discussing offline - this is only for defaults atm, so I dont have a way of cooking up something that fails this atm. Defaults always have a DefaultsSource so far